src/control/cmd/daos_server/storage.go

```diff
@@ -107,13 +107,24 @@ func (cmd *storagePrepareCmd) Execute(args []string) error {
 		}
 	}
 
-	if cmd.ScmOnly || !cmd.NvmeOnly {
+	for cmd.ScmOnly || !cmd.NvmeOnly {
```

Comment:
This is a confusing construction and requires extra thought as to what is happening here. In fact, I've been looking at it for a couple of minutes and I still don't quite understand why it's a loop now.

Suggest refactoring these blocks to be controlled by new booleans, e.g.:
```go
prepScm := cmd.ScmOnly || !cmd.NvmeOnly
prepNvme := cmd.NvmeOnly || !cmd.ScmOnly
```

The benefit here is that there is clarity about when a block will be entered, and it allows you to potentially do other more complex logic to determine when those booleans should be set to true (e.g. probing storage state, etc).

Reply:
it's a loop so can be broken out of